skimage/feature/_hog.py

```diff
@@ -145,8 +128,8 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8),
             for y in range(n_cellsy):
                 for o in range(orientations):
                     centre = tuple([y * cy + cy // 2, x * cx + cx // 2])
-                    dx = radius * cos(float(o) / orientations * np.pi)
-                    dy = radius * sin(float(o) / orientations * np.pi)
+                    dx = radius * np.cos(float(o) / orientations * np.pi)
+                    dy = radius * np.sin(float(o) / orientations * np.pi)
```

Comment:
This visualization loop has always been rather inefficient. We can improve it.

1. Add `from __future__ import division` as the first line of the file so we can eliminate the `float()` calls.
2. Before _any_ of these `for` loops, calculate

``` python
orientations_arr = np.array(orientations)
dx_arr = np.cos(orientations_arr / orientations * np.pi)
dy_arr = np.sin(orientations_arr / orientations * np.pi)
```

Finally, change the final loop to iterate over

``` python
for o, dx, dy in zip(orientations_arr, dx_arr, dy_arr):
```